cid/common.py

```diff
@@ -298,68 +301,104 @@ def _deploy(self, dashboard_id: str=None, recursive=True, update=False, **kwargs
         required_datasets_names = dashboard_definition.get('dependsOn', dict()).get('datasets', list())
 
         dashboard_datasets = dashboard.datasets if dashboard else {}
-        
+
         for name, id in dashboard_datasets.items():
             if id not in self.qs.datasets:
                 logger.info(f'Removing unknown dataset "{name}" ({id}) from dashboard {dashboard_id}')
                 del dashboard_datasets[name]
 
-        # Get QuickSight template details
-        try:
-            source_template = self.qs.describe_template(
-                template_id=dashboard_definition.get('templateId'),
-                account_id=dashboard_definition.get('sourceAccountId'),
-                region=dashboard_definition.get('region', 'us-east-1')
-            )
-        except CidError as exc:
-            raise CidCritical(exc) # Cannot proceed without a valid template
-        dashboard_definition.update({'sourceTemplate': source_template})
-        print(f'\nLatest template: {source_template.arn}/version/{source_template.version}')
-        
-        compatible = self.check_dashboard_version_compatibility(dashboard_id)
-        
+        compatible = True
+        if dashboard_definition.get('templateId'):
+            # Get QuickSight template details
+            try:
+                source_template = self.qs.describe_template(
+                    template_id=dashboard_definition.get('templateId'),
+                    account_id=dashboard_definition.get('sourceAccountId'),
+                    region=dashboard_definition.get('region', 'us-east-1')
+                )
+            except CidError as exc:
+                raise CidCritical(exc) # Cannot proceed without a valid template
+            dashboard_definition['sourceTemplate'] = source_template
+            print(f'\nLatest template: {source_template.arn}/version/{source_template.version}')
+            compatible = self.check_dashboard_version_compatibility(dashboard_id)
+        elif dashboard_definition.get('data'):
```

Comment:
could 'data' key doesn't exist? maybe better to change with elif 'data' in dashboard_definition so it doesn't throw an error?

Reply:
get returns none if data is not in the dict.